Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m

```diff
@@ -510,83 +516,33 @@ - (nullable id)tryToLoadCellMedia:(nullable id (^)(void))loadCellMediaBlock
     cellMedia = loadCellMediaBlock();
     if (cellMedia) {
         DDLogVerbose(@"%@ cell media cache miss", self.logTag);
-        [cellMediaCache setObject:cellMedia forKey:cacheKey];
+        if (!shouldSkipCache) {
+            [cellMediaCache setObject:cellMedia forKey:cacheKey];
+        }
     } else {
         DDLogError(@"%@ Failed to load cell media: %@", [self logTag], [self.attachmentStream mediaURL]);
         self.viewItem.didCellMediaFailToLoad = YES;
-        [mediaView removeFromSuperview];
-        // TODO: We need to hide/remove the media view.
-        [self showAttachmentErrorView];
+        // TODO: Do we need to hide/remove the media view?
+        [self showAttachmentErrorView:mediaView];
     }
     return cellMedia;
 }
 
-// We want to lazy-load expensive view contents and eagerly unload if the
-// cell is no longer visible.
-- (void)ensureViewMediaState
+// * If cell is visible, lazy-load (expensive) view contents.
+// * If cell is not visible, eagerly unload view contents.
```

Comment:
This method got really simple - we now use `loadCellContentBlock` and `unloadCellContentBlock`, so that we can gather concerns and define how to load media views in the same place that we create the media views.